flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -86,11 +88,11 @@ public MetricRegistry(Configuration config) {
 			// by default, create JMX metrics
 			LOG.info("No metrics reporter configured, exposing metrics via JMX");
 			this.reporter = new JMXReporter();
-			this.timer = null;
+			this.executor = null;
 		}
 		else {
 			MetricReporter reporter;
-			java.util.Timer timer;
+			ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Why not instantiating the `ScheduledExecutorService` when we know that the `MetricReporter` is of type `Scheduled`?